internal/upf/upf.go

```diff
@@ -300,3 +303,112 @@ func (u *UPF) listenForTrafficNotifications() {
 		}
 	}
 }
+
+func (u *UPF) monitorUsage(interval time.Duration, stop <-chan struct{}) {
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			err := u.pollUsageAndResetCounters()
+			if err != nil {
+				logger.UpfLog.Warn("Failed to poll usage and reset counters", zap.Error(err))
+			}
+		case <-stop:
+			return
+		}
+	}
+}
+
+func (u *UPF) getUsageForURR(urrID uint32) (uint64, error) {
+	var perCPU []uint64
+
+	err := u.bpfObjects.N3N6EntrypointMaps.UrrMap.Lookup(&urrID, &perCPU)
```

Comment:
This change would simplify the code and reduce race condition risks. We could remove the separate `resetURR`.

It would not be entirely race condition proof, but the alternative would be to add a spinlock to the map.

Combined with the atomic suggestion in the XDP code, it would still be possible for this scenario to happen:

1. User space running on CPU 1 runs the lookup part of LookupAndDelete
2. XDP running on CPU 0 atomically updates the URR
3. User space running on CPU 1 runs the delete part of LookupAndDelete, overwriting the last URR

Combining the operations in LookupAndDelete reduces the time this can happen significantly, and combined with the atomic change in XDP, we would only rarely lose the count of a packet in the URR.

If this is still unacceptable, we will need to add a spinlock to the map.

```suggestion
	err := u.bpfObjects.N3N6EntrypointMaps.UrrMap.LookupAndDelete(&urrID, &perCPU)
```

Reply:
`LookupAndDelete` deletes the key in the map however, which is only created once by the user space program during the PFCP session establishment request.

We get these errors as the second time we run the loop.

```
failed to lookup URR: lookup and delete: key does not exist
```

As far as I know there is no `LookupAndUpdate`.

Reply:
ok, I moved the `Lookup` and `Delete` as close as I could, let me know what you think.